### ISSUE TITLE
Shift+Option+LeftArrow doesn't select word preceding the cursor.

### ISSUE THREAD

**Issue by mitchellpalermo** [Microsoft/vscode#58921]
- VSCode Version: 1.27.2
- OS Version: MacOs 10.13.6

Steps to reproduce
1. Position cursor in front of word to be selected
2. Press Shift+Option+LeftArrow 
3. Word not selected.

Shift+Option+RightArrow functions as expected. The cursorWordStartLeftSelect keybinding is blank. When I try to assign it a value, I can input Shift+Option but when I push the left arrow nothing happens. 

I've tried using the "Define Keybinding" button but it also won't let me input the left arrow. 

I've also tried to add the following custom keybinding to the keybindings.json file: 

  {
        "key": "shift+option+left",
        "command": "cursorWordStartLeftSelect",
        "when": "textInputFocus"
    }

But that doesn't seem to work either. 

I've also looked through my System Preferences to see if there are any conflicting shortcuts and I've found none.

**Comment by alexandrudima** [comment]
@mitchellpalermo Does the keybinding reach vs code? If it does not, then there is nothing we can do about it, you must have some other software that is eating it up. Have you been successful in using this keybinding in other applications on the same machine?

![keyboard-input](https://cloud.githubusercontent.com/assets/5047891/25478333/fe31c830-2b40-11e7-865c-b2c3bb10dd75.gif)

**Comment by mitchellpalermo** [comment]
When I follow the procedure displayed above, I can type in 'Shift+Option+{any other arrow key}'. But it will never allow me to type in the left arrow after the other keys. 

I can use this keyboard shortcut in any other application to select a single word at a time. Do you have any examples of applications that might be blocking the shortcut?

**Comment by alexandrudima** [comment]
If the keybinding is not reaching VS Code, there is nothing we can do on our side.

You can also try:
- Is your `keybindings.json` file empty or did you customize it in any way?
- Does this reproduce when commenting out your settings from `settings.json`?
- Does this reproduce when disabling all extensions?

[ I don't believe this is an issue in VS Code, select word left (`shift+option+left`) is a keybinding used by very many people every day, including myself. ]

**Comment by mitchellpalermo** [comment]
I removed everything in the keybindings.json file and that didn't seem to help.
I also commented out my settings from settings.json. That didn't change anything.
And I disabled all extensions, but to no avail. 

However, I did notice that when I first launch VS Code, I can use the keyboard shortcut for about 7 seconds before the functionality drops out. So for the first 7ish seconds that the app is running, I can use Shift+Option+LeftArrow without issue and then suddenly it stops allowing me to use it. 

I've quit all other applications on my computer and stopped anything from running in the taskbar at the top of the screen. None of that makes a difference. 

What gives?

**Comment by alexandrudima** [comment]
@mitchellpalermo I have no idea, but I still believe it is some kind of environmental issue. The keybindings input does nothing else than display the incoming key event. i.e. there is no state that would change to prohibit that keybinding from being displayed...

Did you try using the Guest account on the mac? I've heard that using that can help to troubleshoot things as that account should have clean OS settings. Have you been successful in using this keybinding in VS Code in the past? Did something change recently for you?

**Comment by mitchellpalermo** [comment]
I tried it with the Guest Account and it worked perfectly fine. 

I don't think anything changed for me recently. I added some extensions, but those don't seem to have an effect on the problem. 

Here's a gif with the keyboard viewer up to show that the keys are being registered by the OS, but not by VS Code. 
![keyboardshortcut](https://user-images.githubusercontent.com/20684298/45918607-15121300-be4f-11e8-9ea6-abc4252602bb.gif)

I've trie re-downloading VSCode but that didn't seem to work either.

**Comment by alexandrudima** [comment]
@mitchellpalermo Let's see if there's something corrupt in VSCode's files: Install https://code.visualstudio.com/insiders/ and try there. That is our nightly build which will use a fresh user data directory, no extensions, etc. Can you use `Shift+Option+Left` in the Insiders version for more than 7 seconds ?

**Comment by mitchellpalermo** [comment]
I tried this and it seems to not work at all! 

Which doesn't make sense. Here's a gif where you can see me opening the Insider's edition and then trying the keyboard shortcut. 

![insider_edit](https://user-images.githubusercontent.com/20684298/45920473-e2c1df00-be69-11e8-8e69-f6facd1c8595.gif)

**Comment by alexandrudima** [comment]
@mitchellpalermo Are you certain this keyboard shortcut works in other software? Does it work in the editor embedded at https://microsoft.github.io/monaco-editor/ ?

**Comment by mitchellpalermo** [comment]
Strangely it works on some apps, but not others. 

It works in Safari and Microsoft Word. 
But not in Chrome or VS Code.
![otherapps](https://user-images.githubusercontent.com/20684298/45920711-2cacc400-be6e-11e8-8424-7676e52f934e.gif)

**Comment by alexandrudima** [comment]
@mitchellpalermo You can use this test page from w3c -- https://www.w3.org/2002/09/tests/keys.html

If Shift+Option+LeftArrow works in Firefox/Safari on that machine and does not work on Chrome, you can create a new issue at https://bugs.chromium.org/p/chromium/issues/list

Because this reproduces on both Chrome and VS Code - both built with Chromium -- I believe this is an upstream issue and should be dealt with by Chromium.

**Comment by mitchellpalermo** [comment]
I tried it out and you're right, I think it's upstream. The shortcut doesn't work in the link that you sent. 

Thanks for continuing to help out with this. It means a lot that you spend time on these things.

**Comment by mitchellpalermo** [comment]
Okay, I've found a fix for this. 

Apparently the Mission Control keyboard shortcut "Move to left a space" being set to "option+left" creates conditions that override text selection in Chromium. "Option+Left" is not the default shortcut. So when I restored the default shortcuts, everything started working again.

I don't remember customizing the shortcuts, but it's possible that I did so many years ago, as I've been carrying this profile from machine to machine for years.

Here's the link to the issue thread that contains the solution: 
https://bugs.chromium.org/p/chromium/issues/detail?id=874219&q=shift%2Boption%2Bleft&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified
